airbyte-ci/connectors/auto_merge/src/auto_merge/main.py

```
@@ -70,11 +76,7 @@ def get_pr_validators(pr: PullRequest) -> set[Callable]:
     Returns:
         list[callable]: The validators
     """
-
-    for label in pr.labels:
-        if label.name in VALIDATOR_MAPPING:
-            return VALIDATOR_MAPPING[label.name]
-    return VALIDATOR_MAPPING[AUTO_MERGE_LABEL]
+    return VALIDATOR_MAPPING[AUTO_MERGE_BYPASS_CI_CHECKS_LABEL]
```

might not be a realistic case, but I believe this has changed the behavior _when a PR has no labels_. Previously, this would have returned they auto-merge label, and now it will return the auto-merge-and-bypass-ci label.

---

maybe this is actually fine because the validator will fail when actually invoked if the label isn't present. I was mildly concerned this would allow more PRs to bypass CI checks than we wanted.

---

@dbgold17 - Thanks very much for the review. I think your instincts are correct here and I've updated the implementation.

1. While I am failure confident the caller is now going to only call this method with the force-merge/ignore-ci-checks use cases, it's safer if we assert it explicitly in case the code changes.
2. I've updated the implementation to hard-fail (ValueError) if called with a PR that doesn't match the expected label.
3. I've left a note that in the future we could fallback to returning an always-false-returning callable, for now I'd rather confirm our assumptions with a hard failure rather than continue without exception.

https://github.com/airbytehq/airbyte/blob/074cb44cdd4201cb7f01d1508bdb387342190b6c/airbyte-ci/connectors/auto_merge/src/auto_merge/main.py#L79-L91